pkg/liveshare/session.go

```diff
@@ -46,17 +52,41 @@ func (s *Session) registerRequestHandler(requestType string, h handler) func() {
 	return s.rpc.register(requestType, h)
 }
 
-// StartsSSHServer starts an SSH server in the container, installing sshd if necessary,
-// and returns the port on which it listens and the user name clients should provide.
-func (s *Session) StartSSHServer(ctx context.Context) (int, string, error) {
+func WithUserPublicKeyFile(path string) func(opts *StartSSHServerOptions) {
+	return func(opts *StartSSHServerOptions) {
+		opts.userPublicKeyFile = path
+	}
+}
+
+// StartSSHServer starts an SSH server in the container, installing sshd if necessary, applies specified
+// options, and returns the port on which it listens and the user name clients should provide.
+func (s *Session) StartSSHServer(ctx context.Context, configureOptions ...func(*StartSSHServerOptions)) (int, string, error) {
```

Comment:
The resulting opts struct has only 1 field. Could this whole construct just be replaced with a regular string argument, or with a simple opts struct at least?
```suggestion
func (s *Session) StartSSHServer(ctx context.Context, userPublicKeyFile string) (int, string, error) {
```

Reply:
There are several other callers of `StartSSHServer` that don't need to send this option (they just don't show up here because they weren't edited). Using the `...func` approach here lets those callers not need to care about the options. Alternatively I could force all the other callers to start passing empty/nil options, or "overload" this method with `StartSSHServer` and `StartSSHServer` with options. WDYT?

Reply:
Chatted with @josebalius and we went with `StartSSHServer` and `StartSSHServerWithOptions`